### DISCUSSION TITLE
How to bind the value of a list item of an object to a label ?

### DISCUSSION THREAD

**Discussion by CrimsonGhost69** [zauberzeug/nicegui#1626] [Q&A] [answered]
### Question

Hello, 

I am having difficulties with the subject of bindings. 

I have a singleton object with a list as class attribute. 
I would like to bind list elements of the object to a label. 

```python
from nicegui import ui


class A():
    _instance = None
    
    inputs = [0,0,0,0,0,0,0,0]

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(A, cls).__new__(cls)
            print("A Instance created.")
        return cls._instance


myObj = A()

ui.label().style('color: black').bind_text_from(myObj, 'inputs[1]')

ui.run()
```

Unfortunately this won't do it. 
Does the binding only work between nicegui objects ?

Many thanks in advance

**Comment by falkoschindler** [comment, marked as answer]
The `target_name` parameter must be the name of the attribute you want to bind against. So specifying an index does not work. But you can use a conversion to pick the element out of the list:
```py
ui.label().style('color: black').bind_text_from(myObj, 'inputs', backward=lambda x: x[1])
```

**Reply by CrimsonGhost69** [reply]
Thank you very much, Falko!
